backend/src/main/java/com/woowacourse/friendogly/club/service/ClubQueryService.java

```diff
@@ -18,23 +23,56 @@
 public class ClubQueryService {
 
     private final ClubRepository clubRepository;
+    private final MemberRepository memberRepository;
+    private final PetRepository petRepository;
 
-    public ClubQueryService(ClubRepository clubRepository) {
+    public ClubQueryService(
+            ClubRepository clubRepository,
+            MemberRepository memberRepository,
+            PetRepository petRepository
+    ) {
         this.clubRepository = clubRepository;
+        this.memberRepository = memberRepository;
+        this.petRepository = petRepository;
     }
 
-    public List<FindSearchingClubResponse> findSearching(FindSearchingClubRequest request) {
+    public List<FindSearchingClubResponse> findSearching(Long memberId, FindSearchingClubRequest request) {
+        Member member = memberRepository.getById(memberId);
+        List<Pet> pets = petRepository.findByMemberId(memberId);
+
         Specification<Club> spec = ClubSpecification.where()
                 .equalsAddress(request.address())
                 .hasGenders(request.genderParams())
                 .hasSizeTypes(request.sizeParams())
                 .build();
 
-        return clubRepository.findAll(spec).stream()
+        List<Club> clubs = clubRepository.findAll(spec);
+
+        if (FilterCondition.toFilterCondition(request.filterCondition()) == FilterCondition.ABLE_TO_JOIN) {
```

Comment:
if문안에 많은 괄호가 있어서 if문의 조건이 흐려지는 것 같아요!
```suggestion
        FilterCondition filterConditon = FilterCondition.toFilterCondition(request.filterCondition());
        if (filterCondition == FilterCondition.ABLE_TO_JOIN) {
        // OR if (filterCondition.isJoinable()) {
```

Reply:
오 enum class 안에 로직을 넣어버리는게 제일 깔끔하겠네요

Reply:
오 트레 피드백 반영하면서, 적용했습니다.